pkg/controller/infinispan/autoscale.go

```diff
@@ -0,0 +1,174 @@
+package infinispan
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	infinispanv1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	"github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Starting a go routine that does polling autoscaling
+func addAutoscalingEquipment(mgr manager.Manager, r reconcile.Reconciler) {
+	go autoscalerPoll(mgr, r)
+}
+
+var defaultMinimumPollPeriod = time.Second
+
+func autoscalerPoll(mgr manager.Manager, r reconcile.Reconciler) {
+	log.Info("Starting adhoc loop for autoscaling")
+	for true {
+		time.Sleep(defaultMinimumPollPeriod)
+		ispnList := infinispanv1.InfinispanList{}
+		// Check all the cluster in the namespace for autoscaling
+		err := mgr.GetClient().List(context.TODO(), &ispnList)
+		if err != nil {
+			log.Error(err, "Unable to select cluster pods list")
+			continue
+		}
+		for _, iItem := range ispnList.Items {
+			if !iItem.IsConditionTrue("wellFormed") {
```

Comment:
@rigazilla looks like we should check that auto-scaling is enabled ASAP early, than cluster is formed to prevent unnecessary work.
Wdyt to move Autoscale and/or any other optional parameters options to the separate CRD and apply it if only needed?

Reply:
@dmvolod yeah thanks! I've also added a check that the cluster is running a CacheService. Since only that is supported atm.

What do you mean with separated CRD?

Reply:
We can create a new resource, for example called InfinispanOptions or InfinispanFeatures or InfinispanControl or anything else and define optional parameters there to keep Infinispan cluster CR simple.
Wdyt?

Reply:
I would try to reach this using the Cache/Infinispan CRDs pair:
- promoting the Cache CR as simple interface for user/application developer
- leaving all the configuration stuff in the Infinispan CR

Actually, I would like to simplify Infinispan CRD itself, but before split data in an external struct I wuold like to try:
- a proper usage of json `omitempty,optional` tags
- a proper definition of optional fields as pointers instead of structs

Reply:
Ok, @rigazilla sure. Let's try to optimize this option.